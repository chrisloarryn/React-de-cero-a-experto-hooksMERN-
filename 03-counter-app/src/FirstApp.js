import React from 'react'
import PropTypes from 'prop-types'

// Functional Component
const FirstApp = ({ message, subtitle = 'My first App' }) => {
  // console.log(message)

  // A LOT OF MEMORY
  // if (!message) {
  //     throw new Error('Message must be specified for the first app component')
  // }

  // eslint-disable-next-line
  const sayHello = {
    message: 'Hello World :D',
    age: 24,
  }
  return (
    <>
      <h1>{message}</h1>
      {/* <pre>{JSON.stringify(sayHello, null, 2)}</pre> */}
      <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  message: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
  subtitle: `I'm a subtitle`,
}

export default FirstApp
