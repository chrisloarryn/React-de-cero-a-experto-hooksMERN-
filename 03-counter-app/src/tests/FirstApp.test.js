import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import FirstApp from '../FirstApp'

/*  
  //   test('<FirstApp /> should render "Hello, I am Goku" message" ', () => {
  //     const greeting = 'Hello, I am Goku'
  //     const { getByText } = render(<FirstApp message={greeting} />)
  //     expect(getByText(greeting)).toBeInTheDocument()
  //   })
*/
describe('Test in <FirstApp />', () => {
  test('<FirstApp /> should be rendered correctly', () => {
    const message = 'Hello, I am Goku'
    const wrapper = shallow(<FirstApp message={message} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('should be return subtitle sended by props', () => {
    const message = 'Hello, I am Goku'
    const subTitle = 'I am a subtitle'
    const wrapper = shallow(<FirstApp message={message} subTitle={subTitle} />)

    const paragraphText = wrapper.find('p').text()
    
    expect(paragraphText).toBe(subTitle)
  })
})
