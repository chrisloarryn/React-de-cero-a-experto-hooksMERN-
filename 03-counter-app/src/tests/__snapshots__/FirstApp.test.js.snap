// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> <FirstApp /> should be rendered correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Goku
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
